central/group/datastore/datastore_impl.go

```diff
@@ -148,7 +148,8 @@ func (ds *dataStoreImpl) Mutate(ctx context.Context, remove, update, add []*stor
 	}
 
 	for _, group := range update {
-		if err := ds.validateAndPrepGroupForUpdateNoLock(ctx, group); err != nil {
+		// We do not allow updates to groups even with the force flag set, hence explicitly setting force to false here.
```

Comment:
This seems surprising to me:
1. Why? Where is this documented? Should the respective traits enum then not be called `ALLOW_FORCE_DELETE_ONLY`?
2. Why does `validateAndPrepGroupForUpdateNoLock` even take a force parameter if it is never set to `true`?

Reply:
I've removed the force parameter for `validateAndPrepGroupForUpdateNoLock`.

Based on what I got from [the conversations](https://github.com/stackrox/stackrox/pull/2610#discussion_r938571896) it seems we have agreed to not support the update for groups for now, simply due to the reason that we don't necessarily have a use-case. If people wish to desire to change the group, they would have to re-create it at the moment. 

Maybe @rukletsov can also share some additional thoughts on this.

Reply:
What would be the use case for force delete in the first place, though? It would seem that making an API call with `ALLOW_FORCE` voids your warranty (i.e., you might soft-brick your instance), so calling it a "use case" seems a bit of a stretch to me in the first place. If we talk about the need to fix things with a broken instance, it seems illogical that we would require a delete and recreate workflow given that "delete" is already nothing that should be exposed to a user.

Reply:
Personally, I don't mind adding the force flag also to the update API, but I know there was something agreed beforehand to not do it (and the reasoning for that was that there's no real _why_ to do this).

I'll add it for now, and once @rukletsov has shared some of the thoughts we can still decide to do it / not do it.

Reply:
Whatever, folks, we're building a kludge here : ). Indeed, I prefer kludges to feel and look like kludges, with as minimal changes as possible. At the same time, it is reasonable to have at least one option to remove a resource object because we're protecting against Murphy here, not Machiavelli. This is my reasoning for allowing force for delete only.